script/utils/sim-sequence.sh

```diff
@@ -0,0 +1,308 @@
+# fork with state overrides disabled.
+#
+# Usage:
+#   ./script/utils/sim-sequence.sh <network> <array-of-task-IDs>
+#
+# Example:
+#   ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+
+# TODO start an anvil fork and run all simulations against the same node.
+# TODO verify the following test cases, push a commit verifying each one has the expected result in CI:
+#   - expected to pass: ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+#   - expected to pass: ./script/utils/sim-sequence.sh eth "021 022 base-003"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 base-003 ink-001 022"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 base-003 ink-001"
+#   - expected to fail: ./script/utils/sim-sequence.sh eth "021 025"
+#   - expected to fail: create a dir called 021-tmp and run ./script/utils/sim-sequence.sh eth "021 022 base-003 ink-001"
+set -euo pipefail
+
+# --- Functions ---
+
+error_exit() {
+  echo "Error: $1" >&2
+  exit 1
+}
+
+# Disable state overrides in Simulation.sol. The only state override currently
+# disabled is the nonce override.
+disable_state_overrides() {
+  # Create a backup
+  local simulation_file="$root_dir/lib/base-contracts/script/universal/Simulation.sol"
+  local backup_file="${simulation_file}.bak"
+  cp "$simulation_file" "$backup_file"
+
+  # Comment out lines containing 'state = addNonceOverride'.
+  awk '/state = addNonceOverride/ { print "        // " $0; next } { print }' "$simulation_file" > "${simulation_file}.tmp"
+
+  # Replace the original file with the modified version, and ensure cleanup on exit.
+  mv "${simulation_file}.tmp" "$simulation_file"
+  trap cleanup EXIT
+}
+
+cleanup() {
+  unset SIMULATE_WITHOUT_LEDGER
+
+  local simulation_file="$root_dir/lib/base-contracts/script/universal/Simulation.sol"
+  local backup_file="${simulation_file}.bak"
+
+  if [[ -f "$backup_file" ]]; then
+    mv "$backup_file" "$simulation_file"
+  fi
+
+  # Kill the anvil fork at the end if it was started by this script
+  if $DESTROY_ANIVIL_AFTER_EXECUTION; then
+    echo "Anvil is still open. To kill it, please use the command below:"
+    echo "ps aux | grep anvil | grep -v grep | awk '{print \$3}' | xargs kill"
+    #TODO: In the future, we should kill the anvil fork here?
+  fi
+}
+createFork() {
+  # Start a fork
+  # check if the port is already open
+  if lsof -Pi :8545 -sTCP:LISTEN -t >/dev/null ; then
+    log_info "Anvil is detected and running on port 8545, we use the current instance of anvil."
+    DESTROY_ANIVIL_AFTER_EXECUTION=false
+  else  
+    log_info "No instance of anvil is detected, starting anvil fork on \"$ANVIL_LOCALHOST_RPC\" by forking $RPC_URL."
+    if [[ -n "$block_number" ]]; then
+      anvil -f $RPC_URL --fork-block-number $block_number >> /tmp/anvil.logs &
+    else
+      anvil -f $RPC_URL >> /tmp/anvil.logs &
+    fi
+    sleep 5
+  fi
+}
+
+NonceDisplayModified(){
```

Comment:
Can be done in a follow up PR, but would be nice to support the chain governor for 3/3 case here

Reply:
open a tracking issue here -> https://github.com/ethereum-optimism/superchain-ops/issues/647